src/main/java/me/ryanhamshire/GriefPrevention/CommandTabCompleter.java

```diff
@@ -0,0 +1,136 @@
+package me.ryanhamshire.GriefPrevention;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.bukkit.Bukkit;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.entity.Player;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+// See notes in GriefPrevention.onTabComplete() regarding why this is static, and not an actual implementation
+// of org.bukkit.command.TabCompleter.
+class CommandTabCompleter
+{
+    private static final String THE_PUBLIC = "public";
+
+    private CommandTabCompleter()
+    {
+        throw new AssertionError("CommandTabCompleter should not be instantiated");
+    }
+
+    static List<String> onTabComplete(CommandSender sender, Command command, String alias, String[] args)
+    {
+        // Assume all completions will be for players, and not console.
+        if (!delegate.isPlayer(sender))
+        {
+            return null;
+        }
+
+        final String label = alias.toLowerCase(Locale.ROOT);
```

Comment:
GP doesn't implement any alias-specific behavior (that I'm aware of at least) and this actually makes GP incompatible with systems like Bukkit's commands.yml adding new aliases. If you use `Command#getName` instead you support all aliases for a command. The lowercasing is a good move though.

Reply:
Brilliant idea; thanks!